src/render_tasks/d3d12_dof_bokeh.hpp

```diff
@@ -0,0 +1,153 @@
+
+			auto& ps_registry = PipelineRegistry::Get();
+			data.out_pipeline = ((D3D12Pipeline*)ps_registry.Find(pipelines::dof_bokeh))->m_native;
+
+			auto source_rt = data.out_source_rt = static_cast<d3d12::RenderTarget*>(fg.GetPredecessorRenderTarget<T>());
+			
+			for (auto frame_idx = 0; frame_idx < versions; frame_idx++)
+			{
+				// Destination
+				{
+					auto cpu_handle = data.out_allocation.GetDescriptorHandle(COMPILATION_EVAL(rs_layout::GetHeapLoc(params::dof_bokeh, params::DoFBokehE::OUTPUT)));
+					d3d12::CreateUAVFromSpecificRTV(n_render_target, cpu_handle, frame_idx, n_render_target->m_create_info.m_rtv_formats[frame_idx]);
+				}
+				// Source
+				{
+					auto cpu_handle = data.out_allocation.GetDescriptorHandle(COMPILATION_EVAL(rs_layout::GetHeapLoc(params::dof_bokeh, params::DoFBokehE::SOURCE)));
+					d3d12::CreateSRVFromSpecificRTV(source_rt, cpu_handle, frame_idx, source_rt->m_create_info.m_rtv_formats[frame_idx]);
+				}
+			}
+		}
+
+		inline void ExecuteDoFBokehTask(RenderSystem& rs, FrameGraph& fg, SceneGraph& sg, RenderTaskHandle handle)
+		{
+			auto& n_render_system = static_cast<D3D12RenderSystem&>(rs);
+			auto& device = n_render_system.m_device;
+			auto& data = fg.GetData<DoFBokehData>(handle);
+			auto n_render_target = fg.GetRenderTarget<d3d12::RenderTarget>(handle);
+			auto frame_idx = n_render_system.GetFrameIdx();
+			auto cmd_list = fg.GetCommandList<d3d12::CommandList>(handle);
+			const auto viewport = n_render_system.m_viewport;
+
+			d3d12::BindComputePipeline(cmd_list, data.out_pipeline);
+
+			const auto camera_cb = static_cast<D3D12ConstantBufferHandle*>(sg.GetActiveCamera()->m_camera_cb);
+
+			Bokeh_CB cb_data;
+			cb_data.m_f_number = sg.GetActiveCamera()->m_f_number;
+			cb_data.m_blades = sg.GetActiveCamera()->m_aperture_blades;
+			cb_data.m_bokeh_poly = sqrt(std::clamp((sg.GetActiveCamera()->m_f_number - 1.8f) / (5.0f - 1.8f), 0.f, 1.f));
+			cb_data.m_enable_dof = sg.GetActiveCamera()->m_enable_dof;
+			
+			data.cb_handle->m_pool->Update(data.cb_handle, sizeof(Bokeh_CB), 0, frame_idx, (uint8_t*)&cb_data);
+
+			d3d12::BindComputeConstantBuffer(cmd_list, data.cb_handle->m_native, 1, frame_idx);
+
+			cmd_list->m_dynamic_descriptor_heaps[D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV]->StageDescriptors(0, 0, 2, data.out_allocation.GetDescriptorHandle());
+
+
+			cmd_list->m_native->ResourceBarrier(1, &CD3DX12_RESOURCE_BARRIER::UAV(data.out_source_rt->m_render_targets[frame_idx % versions]));
+
+			//TODO: numthreads is currently hardcoded to half resolution, change when dimensions are done properly
+			d3d12::Dispatch(cmd_list,
+				static_cast<int>(std::ceil(n_render_system.m_viewport.m_viewport.Width / 16.f)),
+				static_cast<int>(std::ceil(n_render_system.m_viewport.m_viewport.Height / 16.f)),
+				1);
+		}
+
+		inline void DestroyDoFBokeh(FrameGraph& fg, RenderTaskHandle handle, bool resize)
```

Comment:
You can remove the destroy function is if is unused. Make the lambda empty in the add function.